primitives/src/secondary_key.rs

```diff
@@ -364,7 +364,7 @@ pub mod runtime_upgrade {
         type Into = crate::Identity<AccountId>;
 
         fn migrate(self) -> Option<Self::Into> {
-            Some(crate::Identity {
+            Some(Self::Into {
                 roles: self.roles,
                 primary_key: self.primary_key,
                 secondary_keys: self
```

Comment:
N.B: we have `impl<T: Migrate> Migrate for Vec<T>` so you can just do `self.secondary_keys.migrate()?,`. In fact you can really do (where `crate::Identity` is defined):
```rust
/// Identity information.
#[allow(missing_docs)]
#[derive(Encode, Decode, Default, Clone, PartialEq, Debug, Migrate)]
#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
pub struct Identity<AccountId> {
    pub roles: Vec<IdentityRole>,
    pub primary_key: AccountId,
    #[migrate_from(Vec<path::to::old::SecondaryKey<AccountId>>)]
    pub secondary_keys: Vec<SecondaryKey<AccountId>>,
}
```